### PR TITLE
Better item docs

### PR DESCRIPTION
**Pull request by 12AwsomeMan34**
Expands on the previously basic item docs. Now includes creating an `ItemStack`, spawning an `ItemStack` through an `Item` entity, and creating an `ItemStack` from a `BlockState`. If there is anything else that should be added, point such out.

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

File: source/plugin/items/creating.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    import java.util.Optional;
+    
+    public void spawnItem(ItemStack superMegaAwesomeSword, Location<World> spawnLocation) {
+        Extent extent = spawnLocation.getExtent();
+        Optional<Entity> optional = extent
+            .createEntity(EntityTypes.ITEM, spawnLocation.getPosition());
+        if (optional.isPresent()) {
+            Entity item = optional.get();
+            item.offer(Keys.REPRESENTED_ITEM, superMegaAwesomeSword.createSnapshot());
+            extent.spawnEntity(item, Cause.of(EntitySpawnCause.builder()
```
Comment (gabizou) [inline source/plugin/items/creating.rst]: Note that this also requires that it's a `NamedCause.source(SpawnCause)`

**Comment by Inscrutable** [comment]
It looks pretty good to me, nothing serious to quibble about. If the code examples are OK (and not API 3.10+), it's good to go. Gabi's note on 4.0 can wait.

**Comment by 12AwsomeMan34** [comment]
One of my questions is on the spawning bit. It seems users on the forums are pointing out that spawn causes are not really implemented in 3.0 (though the api for it is there). Should we just leave a note somewhere under the spawning examples clearly stating that as of 3.0, it is not implemented and thus they should simply specify the command source or something and tell them that they should update to use spawn causes when possible? Would probably save plugin devs some time.

**Comment by Inscrutable** [comment]
That sounds fair. We are allowed to admit that some stuff isn't implemented - is it absent, or partial? Recommending the API version where it is complete would not be a bad idea, although we may not be able to include a link yet. =)

**Pull request merged**